drivers/serial/uart_stm32.c

```diff
@@ -1685,6 +1690,15 @@ static void uart_stm32_irq_config_func_##index(const struct device *dev)	\
 #define UART_DMA_CHANNEL(index, dir, DIR, src, dest)
 #endif
 
+#if !(HAS_LPUART_1)
+	#define STM32_UART_SINGLE_WIRE(index) \
+		.single_wire = DT_INST_PROP(index, single_wire),
+#else
+	#define STM32_UART_SINGLE_WIRE(index) \
+		.single_wire = false,
+#endif
```

Comment:
can be simply reduced to
```
.single_wire = DT_INST_PROP(index, single_wire),
```

Reply:
Can't follow here. I had trouble with some LPUARTs as they use the same driver, but I did not add the `single-line` property into their bindings _st,stm32-lpuart.yaml_. So compilation fails for each lpuart. To my understanding they do not support this single-line feature, so I didn't want to add it there.

Reply:
In that case, can you try the following ?
```
     .single_wire = DT_INST_PROP_OR(index, single_wire, false),
```

Reply:
Yep, what @erwango suggest is what I actually meant sorry :-)

Reply:
Much nicer solution and CI does not complain!